Visualization/bokeh_holoviews_datashader.ipynb

```diff
@@ -0,0 +1,560 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Part 3: Images\n",
+    "\n",
+    "The next example demonstrates image visualization at the pixel level with datashader."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# First get a sensor image\n",
+    "import lsst.daf.persistence as dafPersist\n",
+    "\n",
+    "butler = dafPersist.Butler('/project/shared/data/Twinkles_subset/output_data_v2')\n",
+    "subset = butler.subset('calexp')\n",
+    "dataid = subset.cache[0]\n",
+    "image = butler.get('calexp', immediate=True, dataId=dataid)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%opts Image  [height=600 width=650]\n",
+    "%%opts Bounds (color='white')\n",
+    "#%%output size=200\n",
+    "\n",
+    "# Make a fake image just for testing purpose\n",
+    "#zz = np.random.poisson(100, size=(4000, 4000))\n",
+    "#xx, yy = np.meshgrid(np.arange(4000), np.arange(4000))\n",
+    "#zz += xx\n",
+    "#bounds=(0, 0, 4000, 4000)   # Coordinate system: (left, bottom, top, right)\n",
+    "#img = hv.Image(zz, bounds=bounds).options(colorbar=True, cmap=bokeh.palettes.Viridis256, logz=True)\n",
+    "\n",
+    "# Use an actual sensor image\n",
+    "bounds_img = (0, 0, image.getDimensions()[0], image.getDimensions()[1])\n",
+    "img = hv.Image(np.log10(image.image.array), bounds=bounds_img).options(colorbar=True, cmap=bokeh.palettes.Viridis256) # Another option is logz=True\n",
+    "\n",
+    "boundsxy = (0, 0, 0, 0)\n",
+    "box = streams.BoundsXY(source=img, bounds=boundsxy)\n",
+    "bounds = hv.DynamicMap(lambda bounds: hv.Bounds(bounds), streams=[box])\n",
+    "\n",
+    "rasterize(img) * bounds"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As with the histograms, it is possible to use interactive callback features on the image plots, such as the selection box."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "box"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here's another version of the image with a tap stream instead of box select."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%opts Image  [height=600 width=650]\n",
+    "%%opts Points (color='white' marker='x' size=20)\n",
+    "\n",
+    "posxy = hv.streams.Tap(source=img, x=0, y=0)\n",
+    "marker = hv.DynamicMap(lambda x, y: hv.Points([(x, y)]), streams=[posxy])\n",
+    "\n",
+    "rasterize(img) * marker"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "'X' marks the spot! What's the value at that location? Execute the next cell to find out."
```

Comment:
Is the 'X' at 0,0? I only see the upper right part of the 'X'. Maybe better to chose a location farther from the boundary? Is there any way to click on the image to specify the 'X' location?

Reply:
Looks like double clicking on the image will select the 'X' location. Cool! Probably want to note this in the md cell above.

Reply:
Yep. Put the marker in the center of the image by default. Added a note in the markdown about needing to click on the image.